### PR TITLE
Implement cargo distribution

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Station.cpp [hunk 1]
```diff
@@ -693,11 +693,43 @@ namespace OpenLoco
...
+    // 0x004929DB
+    void station::updateCargoDistribution()
     {
-        registers regs;
-        regs.ebp = (int32_t)this;
-        call(0x004929DB, regs);
+        WindowManager::invalidate(Ui::WindowType::station, id());
+        WindowManager::invalidate(Ui::WindowType::stationList);
+        bool hasChanged = false;
+        for (auto i = 0; i < max_cargo_stats; ++i)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Station.cpp:702]:
```
../src/OpenLoco/Station.cpp:702:28: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         for (auto i = 0; i < max_cargo_stats; ++i)
                          ~~^~~~~~~~~~~~~~~~~
```

**Comment by AaronVanGeffen** [comment]
Unfortunately, this diverges from master. The differences are very contained to a particular part of the save — presumably the station structs.

First month:
```
Hex Compare
Produced: 18/03/21 1:49:56 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-10.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/cargoDist/autosave_2021-03-18_13-47-24.SV5
2464699 same byte(s)
3 left orphan byte(s)
16 right orphan byte(s)
100 difference byte(s)
```

Second month:
```
Hex Compare
Produced: 18/03/21 1:50:10 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-21.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/cargoDist/autosave_2021-03-18_13-47-35.SV5
2499373 same byte(s)
3 left orphan byte(s)
10 right orphan byte(s)
70 difference byte(s)
```

Third month:
```
Hex Compare
Produced: 18/03/21 1:50:24 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-33.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/cargoDist/autosave_2021-03-18_13-47-46.SV5
2508208 same byte(s)
4 left orphan byte(s)
12 right orphan byte(s)
87 difference byte(s)
```

**Comment by duncanspumpkin** [comment]
let me guess all of the differences were 0xF instead of a different value.

**Review by AaronVanGeffen — APPROVED** [review]
That's it, exactly. All good now.

First month:
```
Hex Compare
Produced: 18/03/21 3:08:09 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-10.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/autosave_2021-03-18_15-07-17.SV5
2464798 same byte(s)
4 difference byte(s)
```

Second month:
```
Hex Compare
Produced: 18/03/21 3:08:43 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-21.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/autosave_2021-03-18_15-07-28.SV5
2499442 same byte(s)
4 difference byte(s)
```

Third month:
```
Hex Compare
Produced: 18/03/21 3:08:55 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/base/autosave_2021-03-18_13-48-33.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/autosave_2021-03-18_15-07-40.SV5
2508294 same byte(s)
5 difference byte(s)
```

**Pull request merged**